File: Stack/232. Implement Queue using Stacks/implementQueueUsingStacks.js
class MyQueue {
  constructor() {
    this._s1 = [];
    this._s2 = [];
  }

  push(x) {
    while (this._s1.length > 0) {
      this._s2.push(this._s1.pop());
    }
    this._s1.push(x);
    while (this._s2.length > 0) {
      this._s1.push(this._s2.pop());
    }
  }

  pop() {
    return this._s1.pop();
  }

  peek() {
    return this._s1[this._s1.length - 1];
  }

  empty() {
    return this._s1.length === 0;
  }
}

File: Stack/232. Implement Queue using Stacks/implement_queue_using_stacks.py
class MyQueue:

    def __init__(self):
        self._s1 = []
        self._s2 = []

    def push(self, x: int) -> None:
        self._s1.append(x)

    def pop(self) -> int:
        if len(self._s2) == 0:
            while len(self._s1) > 0:
                self._s2.append(self._s1.pop())
        return self._s2.pop()

    def peek(self) -> int:
      if len(self._s2) == 0:
          while len(self._s1) > 0:
              self._s2.append(self._s1.pop())
      return self._s2[len(self._s2) - 1]

    def empty(self) -> bool:
        return len(self._s1) == 0 and len(self._s2) == 0
